### PR TITLE
Smoke test npm build

### PR DESCRIPTION
**Pull request by benwarfield-usds**
Added a job to test that the build actually works in a clean working directory, and then made it pass.

### CODE REVIEW

File: .github/workflows/smoketest.yml [hunk 1]
```diff
@@ -61,3 +61,23 @@ jobs:
         run: npm install
       - name: Lint
         run: npm run-script lint
+  build-test:
```
Comment (sharmaneil) [inline .github/workflows/smoketest.yml:64]: I can't add inline comments for code outside the diff, but on line 14 it looks like this is configured for pull requests into `master`. I'm sure github aliases that with `main`

File: .github/workflows/smoketest.yml [hunk 2]
```diff
@@ -61,3 +61,23 @@ jobs:
...
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.npm
+            ./frontend/node_modules
+          key: npm-${{ hashFiles('frontend/package-lock.json', 'frontend/package.json') }}
+      - name: Node setup
+        run: npm install
+      - name: Build release
+        run: npm run-script build
```
Comment (sharmaneil) [inline .github/workflows/smoketest.yml:83]: random thought: long run, is there an ability for this to test the presence of certain env variables in the build, or if those variables have the right values?

**Review by sharmaneil — APPROVED** [review]
looks good! thanks for the updates!

File: .github/workflows/smoketest.yml (hunk 1 above)
Comment (benwarfield-usds) [reply .github/workflows/smoketest.yml:64]: Ooh, good catch! I'm pretty sure that they don't, actually. On the other hand, the pull request target is mostly there for show￼￼— if we just run the test on all pushes to all branches, it doesn't matter if you do or don't have a pull request.

File: .github/workflows/smoketest.yml (hunk 2 above)
Comment (benwarfield-usds) [reply .github/workflows/smoketest.yml:75]: Totally! In that we can run arbitrary code and just have it fail the build by returning a non-zero status. 😁

**Pull request merged**